File: algorithms/detect-html-and-attributes.js
function processHTML(lines) {
  // Combine lines to handle multiline tags
  const htmlContent = lines.join("\n");

  // Remove HTML comments
  const withoutComments = htmlContent.replace(/<!--[\s\S]*?-->/g, "");

  // Regex to match HTML tags with attributes
  const tagRegex = /<(\w+)([^>]*)>/g;
  const attributeRegex = /(\w+)=["']([^"']*)["']/g;

  let match;
  const results = [];

  while ((match = tagRegex.exec(withoutComments)) !== null) {
    const tagName = match[1];
    const attributesStr = match[2].trim();

    results.push(tagName);

    if (attributesStr) {
      let attrMatch;
      while ((attrMatch = attributeRegex.exec(attributesStr)) !== null) {
        const attrName = attrMatch[1];
        const attrValue = attrMatch[2];
        results.push(`-> ${attrName} > ${attrValue}`);
      }
    }
  }

  console.log(results.join("\n"));
}

function detectHTMLAndAttributes(input) {
  const lines = input.trim().split("\n");
  const numLines = parseInt(lines[0], 10);

  if (numLines >= 1 && numLines < 100) {
    const htmlLines = lines.slice(1, numLines + 1);
    return processHTML(htmlLines);
  }

  return "";
}

// For testing with the sample input
const sampleInput = `9
<head>
<title>HTML</title>
</head>
<object type="application/x-flash"
  data="your-file.swf"
  width="0" height="0">
<!-- <param name="movie" value="your-file.swf" /> -->
<param name="quality" value="high"/>
</object>`;

detectHTMLAndAttributes(sampleInput);

File: algorithms/coding-challenges/hackerrank/detect-html-and-attributes.py
import re

def process_html(lines):
    html_content = '\n'.join(lines)

    without_comments = re.sub(r'<!--[\s\S]*?-->', '', html_content)

    tag_regex = r'<(\w+)([^>]*)>'
    attribute_regex = r'(\w+)=["\'](.*?)["\']'

    results = []

    for match in re.finditer(tag_regex, without_comments):
        tag_name = match.group(1)
        attributes_str = match.group(2).strip()

        results.append(tag_name)

        if attributes_str:
            for attr_match in re.finditer(attribute_regex, attributes_str):
                attr_name = attr_match.group(1)
                attr_value = attr_match.group(2)
                results.append(f"-> {attr_name} > {attr_value}")

    print('\n'.join(results))

def detect_html_and_attributes(input_str):
    lines = input_str.strip().split('\n')
    num_lines = int(lines[0])

    if 0 < num_lines < 100:
        html_lines = lines[1:num_lines+1]
        return process_html(html_lines)

    return ''

# For testing with the sample input
sample_input = """9
<head>
<title>HTML</title>
</head>
<object type="application/x-flash"
  data="your-file.swf"
  width="0" height="0">
<!-- <param name="movie" value="your-file.swf" /> -->
<param name="quality" value="high"/>
</object>"""

# def main():
#     # Read the number of lines
#     num_lines = int(input().strip())

#     if 0 < num_lines < 100:
#         # Read the HTML lines
#         html_lines = [input() for _ in range(num_lines)]
#         result = process_html(html_lines)
#         print(result)

# if __name__ == "__main__":
#     main()
